engine/src/execution_graph/logic_controllers/taskflow/distributing_kernel.cpp

```diff
@@ -0,0 +1,96 @@
+#include "distributing_kernel.h"
+#include "utilities/CommonOperations.h"
+
+namespace ral {
+namespace cache {
+
+distributing_kernel::distributing_kernel(std::size_t kernel_id,
```

Comment:
@rommelDB you should be able to have the same function signature for the Constructor as the base kernel class

Reply:
Fixed.